R/zchunk_L252.MACC.R

```diff
@@ -0,0 +1,394 @@
+
+    # Silence package checks
+    AgProductionTechnology <- AgSupplySector <- AgSupplySubsector <- EPA_MACC_Sector <- EPA_region <-
+      GV_year <- MAC_region <- Non.CO2 <- PCT_ABATE <- Process <- Species <- Year <- bio_N2O_coef <-
+      depresource <- emiss.coef <- input.emissions <- mac.control <- mac.reduction <- region <-
+      scenario <- sector <- stub.technology <- subsector <- supplysector <- tax <- tech_change <- year <- NULL
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    A_regions <- get_data(all_data, "emissions/A_regions")
+    A_MACC_TechChange <- get_data(all_data, "emissions/A_MACC_TechChange")
+    GCAM_sector_tech <- get_data(all_data, "emissions/mappings/GCAM_sector_tech")
+    HFC_Abate_GV <- get_data(all_data, "emissions/HFC_Abate_GV")
+    GV_mac_reduction <- get_data(all_data, "emissions/GV_mac_reduction")
+    L152.MAC_pct_R_S_Proc_EPA <- get_data(all_data, "L152.MAC_pct_R_S_Proc_EPA")
+    L201.ghg_res <- get_data(all_data, "L201.ghg_res")
+    L211.AGREmissions <- get_data(all_data, "L211.AGREmissions")
+    L211.AnEmissions <- get_data(all_data, "L211.AnEmissions")
+    L211.AGRBio <- get_data(all_data, "L211.AGRBio")
+    L232.nonco2_prc <- get_data(all_data, "L232.nonco2_prc")
+    L241.hfc_all <- get_data(all_data, "L241.hfc_all")
+    L241.pfc_all <- get_data(all_data, "L241.pfc_all")
+
+    # ===================================================
+    # Prepare the table with all MAC curves for matching
+    # This contains all tax and mac.reduction values
+    L252.MAC_pct_R_S_Proc_EPA <- L152.MAC_pct_R_S_Proc_EPA %>%
+      gather(tax, mac.reduction, matches("[0-9]+")) %>%
+      mutate(tax = as.numeric(tax)) %>%
+      rename(mac.control = Process)
+
+    MAC_taxes <- unique(L252.MAC_pct_R_S_Proc_EPA$tax)
+
+    # This is a function to add in the mac.reduction curves to data
+    # Function needed because these steps are repeated 5 times
+    mac_reduction_adder <- function(df, order, error_no_match = T){
+      df <- df %>%
+        # Add tax values
+        repeat_add_columns(tibble(tax = MAC_taxes)) %>%
+        arrange_("region", order) %>%
+        # Join in EPA regions
+        left_join_error_no_match(A_regions %>%
+                                   select(region, EPA_region = MAC_region),
+                                 by = "region")
+      # Next, add in mac.reduction values
+      if (error_no_match){
+        # Usually we use left_join_error_no_match
+        df <- df %>%
+          left_join_error_no_match(L252.MAC_pct_R_S_Proc_EPA, by = c("EPA_region", "mac.control", "tax"))  %>%
+          mutate(mac.reduction = round(mac.reduction, 3))
+      } else {
+        # There are times where the data does not match, so using left_join is necessary
+        df <- df %>%
+          left_join(L252.MAC_pct_R_S_Proc_EPA, by = c("EPA_region", "mac.control", "tax"))  %>%
+                      mutate(mac.reduction = round(mac.reduction, 3))
+      }
+      return(df)
```

Comment:
Could remove this line and change above lines to `df %>% left_join...`

Reply:
Not sure what you're asking for here. Do you just want to compress lines 103 and 104 into one line?

Reply:
I just meant there's no need for an explicit `return`. But it's fine to leave.